### PR TITLE
Drop duplicate PermissionState definition

### PR DESCRIPTION
**Pull request by foolip**
Fixes https://github.com/w3c/deviceorientation/issues/82.

***
<a href="https://pr-preview.s3.amazonaws.com/foolip/deviceorientation/pull/88.html" title="Last updated on Feb 23, 2021, 2:53 AM UTC (1e5d7a1)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/deviceorientation/88/b95751e...foolip:1e5d7a1.html" title="Last updated on Feb 23, 2021, 2:53 AM UTC (1e5d7a1)">Diff</a>

### CODE REVIEW

**Comment by rakuco** [comment]
Is there anything preventing this from being merged? I recall some long discussions about permissions in #57 and #68. WebKit does not define `PermissionState` in other IDLs, but I guess they can just continue shipping their enum in `DeviceOrientationOrMotionPermissionState.idl`.

**Comment by foolip** [comment]
@reillyeon your check mark is green, so I guess you have the power to merge?

File: index.bs
Comment (reillyeon) [inline index.bs:181]: In specification text an additional value `"default"` is also mentioned. These references should be cleaned up to use `"prompt"` instead.

Comment (saschanaz) [reply index.bs:181]: The "Permission model" section seems also redundant and should be removed.

Comment (foolip) [reply index.bs:181]: Oops, this dropped off my radar, I didn't notice the comments last year.

I made an attempt to refresh the patch now to use terms from https://w3c.github.io/permissions/, but it looks like there is no permission for this in https://w3c.github.io/permissions/#permission-registry?

I think one has to invoke https://w3c.github.io/permissions/#request-permission-to-use with a string, but there is no string like "deviceorientation" for this. What to do?

Comment (anssiko) [reply index.bs:181]: I believe both Blink and WebKit use:

1. DeviceMotionEvent: Requires gyroscope & accelerometer
2. DeviceOrientationEvent: Requires gyroscope & accelerometer & magnetometer

@cdumez via https://bugs.webkit.org/show_bug.cgi?id=221399#c13

Comment (saschanaz) [reply index.bs:181]: Hmm, I don't think Gecko and Chrome require permission for these APIs, per my quick test. https://bugzilla.mozilla.org/show_bug.cgi?id=1536382 implies Safari was implementing this, but I have no Apple device, could you do a quick check?

Comment (reillyeon) [reply index.bs:181]: Chromium grants these permissions by default but provides user controls to disable them. We have not yet implemented the requestPermission() method. This is tracked by [issue 947112](https://bugs.chromium.org/p/chromium/issues/detail?id=947112) and is blocked on some settings UX questions.

I support using the mapping to Permission API names suggested above. In browsers which implement the Permissions API and/or the Generic Sensors API equivalents to these events the consistency should be good for developers. In browsers which do not I don't think it makes a difference.

**Comment by FluorescentHallucinogen** [comment]
Is there something that prevents from merging this?

**Comment by reillyeon** [comment]
> Is there something that prevents from merging this?

There's an unresolved comment regarding some additional specification text that should be updated to keep things consistent.

File: index.bs
Comment (FluorescentHallucinogen) [reply index.bs:181]: @ reillyeon
> In specification text an additional value `"default"` is also mentioned. These references should be cleaned up to use `"prompt"` instead.

I've also found that https://notifications.spec.whatwg.org/#permission-model also uses `"default"` instead of `"prompt"`.

How breaking the changing from `"default"` to `"prompt"`?

(I personally would love to see `"prompt"` instead of `"default"` in both specs.)

Comment (rakuco) [reply index.bs:181]: > How breaking the changing from `"default"` to `"prompt"`?

In this case, I don't think it would be a breaking change because contrary to the Notifications API spec, "default" is not exposed as part of the API here, it's an internal state used in the spec.

Unfortunately, also contrary to the Notifications API spec there's no integration with the Permissions API here whatsoever (see #70). Implementing that integration would allow us to get rid of the "Permission model" section, some of the existing text and likely simplify the `requestPermission()` algorithms, but then we also go back to some of the comments above about what powerful features we'd requesting permission for (and how to avoid breaking WebKit depending on the values we use).

**Comment by reillyeon** [comment]
@rakuco I took another careful look at this and I think that even though the `"default"` value isn't in the `PermissionState` enum from the Permissions API the way that it is used in this spec doesn't _technically_ expose that to script and so it's fine to land this change as is. We can fix it up in a follow-up which adds proper Permissions API integration and so removes all the existing spec text that uses the `"default"` value.

File: index.bs
Comment (reillyeon) [reply index.bs:181]: Let's clean up the `"default"` value in the follow-up which integrates with the Permissions API. It doesn't look technically necessary in this change.

**Review by reillyeon — APPROVED** [review]

**Comment by rakuco** [comment]
Thanks for reviewing it again! Can either you or @anssiko merge the PR? I don't have commit rights to this repository.

**Pull request merged**
